### ISSUE TITLE
can we configure different background images?

### ISSUE THREAD

**Issue by mfrasca** [osm2go/osm2go#22]
it would be nice to show bing, or esri, as background, not just mapnik.

**Comment by DerDakon** [comment]
Do you mean for the project download? You can configure a WMS background for the actual project.

**Comment by mfrasca** [comment]
I have tried to copy JOSM entry for bing into OSM2go: 

- open the menu, 
- choose WMS, 
- choose Import, 
- Add Bing/https://www.bing.com/maps/ (that's the value in JOSM), 
- click on OK and then again on OK.

it starts downloading WMS capabilities, progress bar reaches a number as high as 76000, not always the same, then aborts with a yellow alert "Incomplete/unexpected reply!"

**Comment by mfrasca** [comment]
I'm not sure I understand how it works, but I have the impression that Bing does something different than the rest of the WMS entries.

**Comment by DerDakon** [comment]
Bing is a tileserver, but no WMS server (found that via web search), so what it outputs is not in the format OSM2go can handle. The URL you paste in is just a web package, so &lt;html&gt; is sort of expected to appear.

**Comment by mfrasca** [comment]
right, too bad.  but back to the title question, can we configure a different background in the project view?  like ESRI, maybe?  or that's also a tileserver?
any idea where to start if we wanted to add a tile client in osm2go?

**Comment by DerDakon** [comment]
You can go to the [JOSM wiki](https://josm.openstreetmap.de/wiki/Maps) and scan your region for WMS providers, e.g. [EOx](https://josm.openstreetmap.de/wiki/Maps/Worldwide#EOxcloudlessSentinel-22018) is available worldwide and seems to work flawlessly in OSM2go. Just pick the base URL from the &lt;url&gt; tag (i.e. without all the parameters) and add it.

**Comment by DerDakon** [comment]
A tile client would be something that behaves similar than what wms.cpp has. I think I need to refactor that a bit so it uses less magic with appdata and will return proper data through an object interface.

**Comment by mfrasca** [comment]
I managed to get at least a couple of EOx layers into OSM2go.  one of them does look pretty, but not enough resolution for being of any practical mapping use.  I have enough programming experience (it's my job, and have used VMS, Android, Java, just to mention a few), I might help, but I have not the faintest idea where to start, how to set up the environment, produce an executable, run it in a virtual machine, or install it in my N900.  my computer is a Debian 10 machine.

**Comment by DerDakon** [comment]
The wiki has basic instructions.

**Comment by mfrasca** [comment]
oops, sorry, my fault, I didn't count on it.  I'll check it against Debian and if necessary complete whatever might be missing.  do you have the original difficult-to-find Nokia libraries somewhere?
